geps/gep-713.md

```diff
@@ -560,52 +578,567 @@ ties:
+- Updating one Policy object could cause many affected objects to need to be
+  updated themselves. This sort of fan-out problem can be very bad for apiserver
+  load, particularly if Policy chnges rapidly, there are a lot of objects, or both.
+
+##### Status needs to be namespaced by implementation
+
+Because an object can be affected by multiple implementations at once, any status
+we add must be namespaced by the implementation.
+
+In Route Parent status, we've used the parentRef plus the controller name for this.
+
+For Policy, we can do something similar and namespace by the reference to the
+implementation's controller name.
+
+We can't easily namespace by the originating Policy because the source could be
+more than one Policy object.
+
+##### Creating common data representation patterns
+
+The problem here is that we need to have a _common_ pattern for including the
+details of an _arbitrarily defined_ object, that needs to be included in the base
+API.
+
+So we can't use structured data, because we have no way of knowing what the
+structure will be beforehand.
+
+This suggests that we need to use unstructured data for representing the main
+body of an arbitrary Policy object.
+
+Practically, this will need to be a string representation of the YAML form of the
+body of the Policy object (absent the metadata part of every Kubernetes object).
+
+Policy Attachment does not mandate anything about the design of the object's top
+level except that it must be a Kubernetes object, so the only thing we can rely
+on is the presence of the Kubernetes metadata elements: `apiVersion`, `kind`,
+and `metadata`.
+
+A string representation of the rest of the file is the best we can do here.
+
+##### Fanout status update problems
+
+The fanout problem is that, when an update takes place in a single object (a
+Policy, or an object with a Policy attached), an implementation may need to
+update _many_ objects if it needs to place details of what Policy applies, or
+what the resultant set of policy is on _every_ object.
+
+Historically, this is a risky strategy and needs to be carefully applied, as
+it's an excellent way to create apiserver load problems, which can produce a large
+range of bad effects for cluster stability.
+
+This does not mean that we can't do anything at all that affects multiple objects,
+but that we need to carefully consider what information is stored in status so 
+that _every_ Policy update does not require a status update.
+
+#### Solution summary
+
+Because Policy Attachment is a pattern for APIs, not an API, and needs to address
+all the problems above, the strategy this GEP proposes is to define a range of
+options for increasing the discoverabilty of Policy resources, and provide
+guidelines for when they should be used.
+
+It's likely that at some stage, the Gateway API CRDs will include some Policy
+resources, and these will be designed with all these discoverabiity solutions
+in mind.
+
+
+### Solution cookbook
```

Comment:
Instead of expanding an already massive GEP, I think I'd prefer to link out to GEPs, issues, discussions, etc for potential solutions.

Reply:
See a below comment, but I think that breaking this up should be a separate piece of work (that I'll pick up straight away).